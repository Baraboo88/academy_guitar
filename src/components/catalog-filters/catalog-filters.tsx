import React, {useCallback, useEffect, useState} from 'react';
import {
  ENTER_KEY,
  getCyrillicTypeFiler,
  getPriceWithSpaces, MAX_PRICE_INITIAL_VALUE,
  MIN_PRICE_INITIAL_VALUE,
  Query,
  QueryModel
} from '../../utils/utils';
import {StateModel} from '../../types/redux-models';
import {

  getGuitarsSelectedMaxPrice,
  getGuitarsSelectedMinPrice,
  getGuitarsSelectedStrings,
  getGuitarsSelectedTypes,

  getGuitarsStrings,
  getGuitarsTypes,
  getMinMaxPrice

} from '../../store/guitars/guitars-selectors';
import {GuitarsActionCreator} from '../../store/guitars/guitars-reducer';
import { GuitarStringCount, GuitarType} from '../../types/guitar-model';
import {connect} from 'react-redux';
import {useNavigate} from 'react-router-dom';
import {useQuery} from '../../hooks/use-query/use-query';

import * as queryString from 'query-string';


interface CatalogFilterProps {
  setMinPrice: (minPrice: number) => void,
  setMaxPrice: (maxPrice: number) => void;
  setGuitarsTypes: (guitarTypes: GuitarType []) => void;
  setGuitarsStrings: (noOfStrings: GuitarStringCount []) => void;
  availableMinMaxPrices: number [],
  availableGuitarsString: GuitarStringCount [],
  availableGuitarsTypes: GuitarType [],
  selectedTypes: GuitarType [];
  selectedStrings: GuitarStringCount [];
  onInnerQuerySet: (newQuery: QueryModel) => void;
  innerQuery: QueryModel;
  selectedMaxPrice: number;
  selectedMinPrice: number;
}


function CatalogFilters(props: CatalogFilterProps) {
  const {
    setMinPrice,
    setMaxPrice,

    availableMinMaxPrices,
    availableGuitarsString,
    availableGuitarsTypes,
    innerQuery,
    onInnerQuerySet,
    selectedTypes,
    selectedStrings,
    setGuitarsTypes,
    setGuitarsStrings,
    selectedMinPrice,
    selectedMaxPrice,
  } = props;

  const [innerMinPrice, setInnerMinPrice] = useState('');
  const [innerMaxPrice, setInnerMaxPrice] = useState('');

  const navigate = useNavigate();
  const query = useQuery();

  const minPriceFromQuery = query.get(Query.MinPrice);
  const maxPriceFromQuery = query.get(Query.MaxPrice);
  const guitarsTypesFromQuery = query.get(Query.GuitarTypes);
  const guitarStringsFromQuery = query.get(Query.GuitarStrings);
  useEffect(()=> {
    if(minPriceFromQuery){
      let newMinPrice = Number(minPriceFromQuery);

      if(newMinPrice < availableMinMaxPrices[0]){
        newMinPrice = availableMinMaxPrices[0];
      }
      if(newMinPrice > availableMinMaxPrices[1]){
        newMinPrice = availableMinMaxPrices[1];
      }
      setMinPrice(newMinPrice);
    } else {
      setMinPrice(MIN_PRICE_INITIAL_VALUE);
    }


  }, [minPriceFromQuery, setMinPrice, availableMinMaxPrices, onInnerQuerySet]);


  useEffect(() => {
    if(Number(innerMaxPrice) !== selectedMaxPrice) {

      onInnerQuerySet({maxPrice:selectedMaxPrice=== MAX_PRICE_INITIAL_VALUE ? MAX_PRICE_INITIAL_VALUE : Number(selectedMaxPrice)});
      setInnerMaxPrice(selectedMaxPrice.toString());
    }
  }, [selectedMaxPrice]);

  useEffect(() => {
    if(Number(innerMinPrice) !== selectedMinPrice) {
      onInnerQuerySet({minPrice:selectedMinPrice=== MIN_PRICE_INITIAL_VALUE ? MIN_PRICE_INITIAL_VALUE : Number(innerMinPrice)});
      setInnerMinPrice(selectedMinPrice.toString());
    }


  }, [selectedMinPrice]);

  useEffect(()=> {
    if(maxPriceFromQuery){
      let newMaxPrice = Number(maxPriceFromQuery);
      if(newMaxPrice !== selectedMaxPrice){
        if(newMaxPrice > availableMinMaxPrices[1]){
          newMaxPrice = availableMinMaxPrices[1];
        }
        if(newMaxPrice < availableMinMaxPrices[0]){
          newMaxPrice = availableMinMaxPrices[0];
        }
        setMaxPrice(newMaxPrice);
      } else {
        setMaxPrice(MAX_PRICE_INITIAL_VALUE);
      }

    }


  }, [maxPriceFromQuery, setMaxPrice, availableMinMaxPrices, onInnerQuerySet]);


  useEffect(() => {

    if(guitarsTypesFromQuery){
      const newGuitarTypes = guitarsTypesFromQuery.split(',').map((el) => el  as GuitarType);

      if(selectedTypes.length !== newGuitarTypes.length){

        onInnerQuerySet({guitarTypes: newGuitarTypes});
        setGuitarsTypes(newGuitarTypes);
      }

    } else {
      if(selectedTypes.length !== 0){
        onInnerQuerySet({guitarTypes: []});
        setGuitarsTypes([]);
      }
    }
  },[selectedTypes, guitarsTypesFromQuery, onInnerQuerySet, setGuitarsTypes]);

  useEffect(() => {
    if(guitarStringsFromQuery){
      const newGuitarStrings = guitarStringsFromQuery.toString().split(',').map((el) => el  as GuitarStringCount);
      if(selectedStrings.length !== newGuitarStrings.length){
        setGuitarsStrings(newGuitarStrings);
        onInnerQuerySet({ guitarStrings: newGuitarStrings});
      }

    } else {
      if(selectedStrings.length !== 0){
        onInnerQuerySet({guitarStrings: []});
        setGuitarsStrings([]);
      }
    }
  },[selectedStrings, guitarStringsFromQuery, setGuitarsStrings,onInnerQuerySet]);

  const enterFunction = useCallback((event) => {

    if (event.key === ENTER_KEY) {
      let minPrice = '';
      let maxPrice = '';

      if(innerMinPrice && Number(innerMinPrice) >= 0) {
        minPrice = innerMinPrice;
      }
      if( innerMaxPrice && Number(innerMaxPrice) >= Number(innerMinPrice)){
        maxPrice = Number(innerMaxPrice) >= Number(innerMinPrice) ? innerMaxPrice :innerMinPrice;
      } else {
        setInnerMaxPrice('');
      }
      if(selectedMinPrice !== Number(innerMinPrice)){
        setMinPrice(MIN_PRICE_INITIAL_VALUE);
      }
      if(selectedMaxPrice !== Number(innerMaxPrice)){
        setMaxPrice(MAX_PRICE_INITIAL_VALUE);
      }


      navigate(generateMinMaxPriceLink(minPrice, maxPrice));


    }
  }, [innerMinPrice, innerMaxPrice, navigate]);


  useEffect(() => {
    document.addEventListener('keydown', enterFunction, false);

    return () => {
      document.removeEventListener('keydown', enterFunction, false);

    };
  }, [enterFunction]);

  const generateMinMaxPriceLink = (minPrice: string, maxPrice: string) => `?${queryString.stringify({...innerQuery, minPrice, maxPrice},  {skipEmptyString: true, arrayFormat: 'comma'})}`;

  const generateGuitarsTypeLink = (elements:GuitarType []) => `?${queryString.stringify({...innerQuery, guitarTypes: elements},  {skipEmptyString: true, arrayFormat: 'comma'})}`;

  const generateGuitarsStringsLink = (elements:GuitarStringCount []) => `?${queryString.stringify({...innerQuery, guitarStrings: elements},  {skipEmptyString: true, arrayFormat: 'comma'})}`;


  return (
    <form className="catalog-filter">
      <h2 className="title title--bigger catalog-filter__title">Фильтр</h2>
      <fieldset className="catalog-filter__block">
        <legend className="catalog-filter__block-title">Цена, ₽</legend>
        <div className="catalog-filter__price-range">
          <div className="form-input">
            <label className="visually-hidden">Минимальная цена</label>
            <input value={Number(innerMinPrice) <= -1 ? '' : innerMinPrice} onChange={(evt) => {
              setInnerMinPrice(evt.target.value);
            }} type="number" placeholder={`${availableMinMaxPrices.length > 0 ? getPriceWithSpaces(availableMinMaxPrices[0]) : ''}`} id="priceMin" name="от"
            />
          </div>
          <div className="form-input">
            <label className="visually-hidden">Максимальная цена</label>
            <input value={Number(innerMaxPrice) <= -1  ? '' : innerMaxPrice} onChange={(evt) => {
              setInnerMaxPrice(evt.target.value);
            }}
            type="number" placeholder={`${availableMinMaxPrices.length > 1 ? getPriceWithSpaces(availableMinMaxPrices[1]) : ''}`} id="priceMax" name="до"
            />
          </div>
        </div>
      </fieldset>
      <fieldset className="catalog-filter__block">
        <legend className="catalog-filter__block-title">Тип гитар</legend>
        {Object.values(GuitarType).map((guitarType, index) =>(
          <div key={guitarType} className="form-checkbox catalog-filter__block-item">
            <input onChange={() => {
              const selectedIndex = selectedTypes.indexOf(guitarType);
              const newSelectedGuitarsTypes = [...selectedTypes];
              if(selectedIndex >= 0){
                newSelectedGuitarsTypes.splice(selectedIndex, 1);
              } else {
                newSelectedGuitarsTypes.push(guitarType);
              }
              navigate(generateGuitarsTypeLink(newSelectedGuitarsTypes));
            }} checked={selectedTypes.includes(guitarType)} className="visually-hidden" type="checkbox" id={guitarType} name={guitarType} disabled={!availableGuitarsTypes.includes(guitarType)}
            />
            <label htmlFor={guitarType}>{getCyrillicTypeFiler(guitarType)}</label>
          </div>),

        )}

      </fieldset>
      <fieldset className="catalog-filter__block">
        <legend className="catalog-filter__block-title">Количество струн</legend>
        {Object.values(GuitarStringCount).filter((value ) => !isNaN(Number(value))).map((stringNo) =>
          (
            <div key={stringNo} className="form-checkbox catalog-filter__block-item">
              <input onChange={() => {
                const selectedIndex = selectedStrings.indexOf(stringNo);
                const newSelectedGuitarsStrings = [...selectedStrings];
                if(selectedIndex >= 0){
                  newSelectedGuitarsStrings.splice(selectedIndex, 1);
                } else {
                  newSelectedGuitarsStrings.push(stringNo);
                }
                navigate(generateGuitarsStringsLink(newSelectedGuitarsStrings));
              }} className="visually-hidden" type="checkbox"
              id={`${stringNo}-strings`} name={`${stringNo}-strings`}
              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
              // @ts-ignore
              disabled={!availableGuitarsString.includes(Number(stringNo))}
              />
              <label htmlFor={`${stringNo}-strings`}>{stringNo}</label>
            </div>))}

      </fieldset>
      <button onClick={() => {

        navigate(`?${queryString.stringify({...innerQuery, guitarTypes: [], guitarStrings: [], maxPrice:'', minPrice:''},  {skipEmptyString: true})}`);
      }} className="catalog-filter__reset-btn button button--black-border button--medium"
      type="reset"
      >
          Очистить
      </button>
    </form>
  );
}

const mapStateToProps = (state: StateModel) => ({
  availableMinMaxPrices: getMinMaxPrice(state),
  availableGuitarsString: getGuitarsStrings(state),
  availableGuitarsTypes: getGuitarsTypes(state),
  selectedMinPrice: getGuitarsSelectedMinPrice(state),
  selectedMaxPrice: getGuitarsSelectedMaxPrice(state),
  selectedTypes: getGuitarsSelectedTypes(state),
  selectedStrings: getGuitarsSelectedStrings(state),
});

/* eslint-disable  @typescript-eslint/no-explicit-any */
const mapDispatchToProps = (dispatch: any) => ({
  setMinPrice(minPrice: number) {
    dispatch(GuitarsActionCreator.setMinPrice(minPrice));
  },
  setMaxPrice(maxPrice: number) {
    dispatch(GuitarsActionCreator.setMaxPrice(maxPrice));
  },
  setGuitarsTypes(guitarTypes: GuitarType []) {
    dispatch(GuitarsActionCreator.setGuitarsTypes(guitarTypes));
  },
  setGuitarsStrings(noOfStrings: GuitarStringCount []) {
    dispatch(GuitarsActionCreator.setNoOfStrings(noOfStrings));
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(CatalogFilters);

export {CatalogFilters};
