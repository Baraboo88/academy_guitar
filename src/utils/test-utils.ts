import {AddCommentModel, GuitarCommentModel, GuitarModel, GuitarType} from '../types/guitar-model';
import {createApi} from '../api';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {ReactWrapper} from 'enzyme';


export const MOCK_ERROR= 'Somme error';

export const mockGuitars: GuitarModel[] = [
  {
    id: 1,
    name: 'Честер Bass',
    vendorCode: 'SO757575',
    type: 'electric' as GuitarType,
    description: 'Замечательный малобюджетный вариант, созданный для новичков, которые отдают предпочтение мелодичным стилям. Прекрасно звучат блюз и баллады, исполненные на этой гитаре. Акустические свойства весьма высоки, в отличие от ее стоимости.',
    previewImg: 'img/guitar-1.jpg',
    stringCount: 7,
    rating: 4,
    price: 17500,
  },
  {
    id: 2,
    name: 'CURT Z300',
    vendorCode: 'TK129049',
    type: 'electric' as GuitarType,
    description: 'Эргономичность гитары и качество сборки являются, пожалуй, её главными преимуществами. Идеальное расположение в руках музыканта дополняется прочностью конструкции из клёна.',
    previewImg: 'img/guitar-8.jpg',
    stringCount: 7,
    rating: 3.5,
    price: 29500,
  },
  {
    id: 3,
    name: 'Roman LX',
    vendorCode: 'RO111111',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле класса премиум от компании CURT, собравшая в себе все самые необходимые качесва: лёгкость корпуса, прочность струн и компактный размер.',
    previewImg: 'img/guitar-6.jpg',
    stringCount: 4,
    rating: 4,
    price: 6800,
  },
  {
    id: 4,
    name: 'CURT T300',
    vendorCode: 'TK436457',
    type: 'electric' as GuitarType,
    description: 'CURT T300 - это шестиструнная электрогитара популярной линейки FPT. Модель с классическим стилем головы грифа и деки.',
    previewImg: 'img/guitar-3.jpg',
    stringCount: 6,
    rating: 5,
    price: 30000,
  },
  {
    id: 5,
    name: 'Dania Super',
    vendorCode: 'DI192138',
    type: 'acoustic' as GuitarType,
    description: 'Гитары производителя Dania пользуются популярностью у музыкантов разного уровня. В модели Super идеально сочетаются демократичная цена и качество. Корпус, выполненный из тополя позволяет корпусу долгое время сохранять свой первоначальный вид.',
    previewImg: 'img/guitar-4.jpg',
    stringCount: 7,
    rating: 4.5,
    price: 3500,
  },
  {
    id: 6,
    name: 'Честер WX',
    vendorCode: 'SO934345',
    type: 'electric' as GuitarType,
    description: 'Электрогитара с олд-скульным грифом и глянцевым корпусом для настоящих рок-музыкантов. Насыщенный звук идеально подойдет для исполнения рок-композиций.',
    previewImg: 'img/guitar-2.jpg',
    stringCount: 6,
    rating: 3.5,
    price: 15300,
  },
  {
    id: 7,
    name: 'Dania VX',
    vendorCode: 'DI082347',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле - прекрасный компактный аналог классической акустической гитары. Прекрасное качество продуктов компании Dania сочетается с демократичными ценами продуктов линейки GSR.',
    previewImg: 'img/guitar-5.jpg',
    stringCount: 4,
    rating: 2.5,
    price: 2200,
  },
];
export const testInitialState = {
  currentGuitar: {
    currentGuitar: mockGuitars,
    errorMsg: '',
  },
  guitars: {
    guitars: mockGuitars,
    errorMsg: '',
  },
};

export const mockCommentToAdd: AddCommentModel = {
  advantage: 'Mock Advantage', comment: 'Mock Comment', disadvantage: 'Mock Disadvantage', guitarId: 3, rating: 3, userName: 'Some name',
};


export const mockComment: GuitarCommentModel = {id: '31b10247-adff-44a7-937e-c71af64f425b',
  userName: 'Никита',
  advantage: 'Цена.',
  disadvantage: 'Быстро приходит в негодность покрытие краской.',
  comment: 'Хорошая гитара для начинающих, сделана качественно. Лучше за эти деньги не найти.',
  rating: 5,
  createAt: new Date('2022-03-10T10:56:01.590Z'),
  guitarId: 1};

export const getTestStore = () => {

  const reducer = (state = testInitialState) => state;
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};

export const findByTestAtr = (component:ReactWrapper, attr: string) => component.find(`[data-test='${attr}']`);
