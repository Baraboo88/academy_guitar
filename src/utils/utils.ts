const STARTS_AMOUNT = 5;

export const imageAdapter = (img: string) => img.slice(img.length - 5, img.length - 4);


export const generateStarsArray = () => {
  const stars = [];
  for (let i = 1; i <= STARTS_AMOUNT; i++) {
    stars.push(i);
  }
  return stars;
};

export const getCyrillicRating = (rating: number) => {
  if(rating> 0 && rating < 3){
    return 'Плохо';
  } else if (rating >= 3 && rating < 4){
    return 'Удовлетворительно';
  } else if(rating >= 4 && rating < 5){
    return 'Хорошо';
  } else {
    return 'Отлично';
  }
};
