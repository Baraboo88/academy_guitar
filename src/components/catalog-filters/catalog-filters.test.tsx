
import * as Enzyme from 'enzyme';

import { mount } from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';

import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';

import {getTestStore} from '../../utils/test-utils';

import { BrowserRouter } from 'react-router-dom';
import {CatalogFilters} from './catalog-filters';
import {MAX_PRICE_INITIAL_VALUE, MIN_PRICE_INITIAL_VALUE} from '../../utils/utils';


Enzyme.configure({adapter: new EnzymeReactAdapter()});


it('CatalogFilters successfully rendered', () => {
  const mockSetMinPrice = jest.fn();
  const mockSetMaxPrice = jest.fn();
  const mockSetGuitarsTypes = jest.fn();
  const mockSetGuitarsStrings = jest.fn();
  const mockOnInnerQuerySet = jest.fn();
  const query = {  page: '',
    sort: '',
    minPrice: '',
    maxPrice: ''};

  const tree = mount(
    <Provider store={getTestStore()}>
      <BrowserRouter>
        <CatalogFilters
          setMinPrice ={mockSetMinPrice}
          setMaxPrice={mockSetMaxPrice}
          setGuitarsTypes={mockSetGuitarsTypes}
          setGuitarsStrings={mockSetGuitarsStrings}
          availableMinMaxPrices={[-1, -1]}
          availableGuitarsString={[]}
          availableGuitarsTypes={[]}
          selectedTypes={[]}
          selectedStrings={[]}
          onInnerQuerySet={mockOnInnerQuerySet}
          innerQuery={query}
          selectedMinPrice={MIN_PRICE_INITIAL_VALUE}
          selectedMaxPrice={MAX_PRICE_INITIAL_VALUE}
        />
      </BrowserRouter>
    </Provider>,
  );
  expect(toJson(tree, {mode: 'deep'})).toMatchSnapshot();
});
