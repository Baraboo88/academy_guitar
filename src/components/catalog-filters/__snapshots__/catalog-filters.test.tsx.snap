// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogFilters successfully rendered 1`] = `
<form
  className="catalog-filter"
>
  <h2
    className="title title--bigger catalog-filter__title"
  >
    Фильтр
  </h2>
  <fieldset
    className="catalog-filter__block"
  >
    <legend
      className="catalog-filter__block-title"
    >
      Цена, ₽
    </legend>
    <div
      className="catalog-filter__price-range"
    >
      <div
        className="form-input"
      >
        <label
          className="visually-hidden"
        >
          Минимальная цена
        </label>
        <input
          id="priceMin"
          name="от"
          onChange={[Function]}
          placeholder="-1"
          type="number"
          value=""
        />
      </div>
      <div
        className="form-input"
      >
        <label
          className="visually-hidden"
        >
          Максимальная цена
        </label>
        <input
          id="priceMax"
          name="до"
          onChange={[Function]}
          placeholder="-1"
          type="number"
          value=""
        />
      </div>
    </div>
  </fieldset>
  <fieldset
    className="catalog-filter__block"
  >
    <legend
      className="catalog-filter__block-title"
    >
      Тип гитар
    </legend>
    <div
      className="form-checkbox catalog-filter__block-item"
      key="acoustic"
    >
      <input
        checked={false}
        className="visually-hidden"
        disabled={true}
        id="acoustic"
        name="acoustic"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="acoustic"
      >
        Акустические гитары
      </label>
    </div>
    <div
      className="form-checkbox catalog-filter__block-item"
      key="electric"
    >
      <input
        checked={false}
        className="visually-hidden"
        disabled={true}
        id="electric"
        name="electric"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="electric"
      >
        Электрогитары
      </label>
    </div>
    <div
      className="form-checkbox catalog-filter__block-item"
      key="ukulele"
    >
      <input
        checked={false}
        className="visually-hidden"
        disabled={true}
        id="ukulele"
        name="ukulele"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="ukulele"
      >
        Укулеле
      </label>
    </div>
  </fieldset>
  <fieldset
    className="catalog-filter__block"
  >
    <legend
      className="catalog-filter__block-title"
    >
      Количество струн
    </legend>
    <div
      className="form-checkbox catalog-filter__block-item"
      key="4"
    >
      <input
        checked={false}
        className="visually-hidden"
        disabled={true}
        id="4-strings"
        name="4-strings"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="4-strings"
      >
        4
      </label>
    </div>
    <div
      className="form-checkbox catalog-filter__block-item"
      key="6"
    >
      <input
        checked={false}
        className="visually-hidden"
        disabled={true}
        id="6-strings"
        name="6-strings"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="6-strings"
      >
        6
      </label>
    </div>
    <div
      className="form-checkbox catalog-filter__block-item"
      key="7"
    >
      <input
        checked={false}
        className="visually-hidden"
        disabled={true}
        id="7-strings"
        name="7-strings"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="7-strings"
      >
        7
      </label>
    </div>
    <div
      className="form-checkbox catalog-filter__block-item"
      key="12"
    >
      <input
        checked={false}
        className="visually-hidden"
        disabled={true}
        id="12-strings"
        name="12-strings"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="12-strings"
      >
        12
      </label>
    </div>
  </fieldset>
  <button
    className="catalog-filter__reset-btn button button--black-border button--medium"
    onClick={[Function]}
    type="reset"
  >
    Очистить
  </button>
</form>
`;
